src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -336,10 +360,16 @@ void AggregatingStep::transformPipeline(QueryPipelineBuilder & pipeline, const B
             /// Do merge of aggregated data in parallel.
             pipeline.resize(merge_threads);
 
-            pipeline.addSimpleTransform([&](const Block &)
+            const auto & required_sort_description = memoryBoundMergingWillBeUsed() ? group_by_sort_description : SortDescription{};
+            pipeline.addSimpleTransform(
```

Comment:
Why transform is added if `memoryBoundMergingWillBeUsed()` is false?

Reply:
I mean if `required_sort_description` is empty we can use just `pipeline.resize(1)`, can't we?

Reply:
it is needed anyway to merge the data